www/src/templates/template-blog-post.js

```diff
@@ -12,17 +12,35 @@ import Container from "../components/container"
 import EmailCaptureForm from "../components/email-capture-form"
 import TagsSection from "../components/tags-section"
 import HubspotForm from "../components/hubspot-form"
+import Pullquote, { PULLQUOTE_CLASSNAME } from "../components/shared/pullquote"
 import Chart from "../components/chart"
 
 const renderAst = new rehypeReact({
   createElement: React.createElement,
   components: {
     "hubspot-form": HubspotForm,
     "date-chart": Chart,
+    pullquote: Pullquote,
   },
 }).Compiler
 
 class BlogPostTemplate extends React.Component {
+  componentDidMount = () => {
```

Comment:
This seems not quite ideal. Can't you just re-use `nth-child` here?

Here's an [example](https://codesandbox.io/s/p32pp5xy4j)

Reply:
I would love to avoid adding classes but for me (now) it's not possible.

#### 1. 
In the example there are no other elements in the container than  `<p><span>text</span></p>`.
```
<div>
  <p><span>text</span></p>
  <p><span>text</span></p>
  <p><span>text</span></p>
</div>
```
in the Gatsby blog we have inside the container also other elements, not only these with `Pullquotes`

```
<div>
  <p>text</p>
  <p>text</p>
  <p><blockquote>text</blockquote></p>
  <p>text</p>
  <p><blockquote>text</blockquote></p>
  <p>text</p>
</div>
```
And we have no idea where the items `<p><blockquote>text</blockquote></p>` will be placed, on which position, so we can't base styling on `p:nth-child(${index + 1}n)`

#### 2. 
We also can't use `:nth-of-type`, which I had in mind when I proposed this 'style variations' feature. Because every of our `<blockquote>` element is always on the first position, it's always the first child of type inside the containing `<p>` tag.

#### 3. 
Assume that we can remove the `<p>` tags containing our `<blockquote>` elements. I would love to have it that way, but I have no idea how to make it (at least for now), probably some logic in `gatsby-transformer-remark` could make it. But even if we would have cleaner markdown like these

```
<div>
  <p>text</p>
  <p>text</p>
  <blockquote>text</blockquote>
  <p>text</p>
  <blockquote>text</blockquote>
  <p>text</p>
</div>
```  
we can't omit the class names, because we cannot touch existing `<blockquote>` elements created with markdown blockquote formatting  `> text...`. So we have to differentiate the `Pullquote` blockquotes from the  generic ones.   

#### 4.
I'm sorry, I'm not sure what do you mean by that 
> If worse came to worse, I almost think a local variable tracking component mounts (and assigning a class) would perhaps be improved, as well.

Could you please explicate it a little bit for me?

Reply:
> in the Gatsby blog we have inside the container also other elements, not only these with Pullquotes

I'm not sure this matters! I've updated my example to show a similar structure. https://codesandbox.io/s/p32pp5xy4j

> Could you please explicate it a little bit for me?

I mean something like this. If the idea is that we want to track instances/occurrences, we can use a local variable (a side-effect) and track it like so in [this example](https://codesandbox.io/s/vmnr47mx57). Note that this is _global_ so it tracks number of invocations of the component. So it wouldn't necessarily be the same every page, e.g. if you went to a page with one pullquote, navigating to a second page would start on the 2nd style.

This _could_ be worse, but I want to give you some options, because I think my "spidey sense" is going off a little bit with the current approach! Totally fine with it as is, just want to make sure we at least vet some other options!

Reply:
> I'm not sure this matters! I've

Actually the updated example is less clear than the original was :). But  they seam to  work only because you generate the list of paragraph, so there are repeating patterns. 

I've simplified the example https://codesandbox.io/s/1op0l5535q

![screenshot from 2018-11-30 06-40-25](https://user-images.githubusercontent.com/32480082/49270731-f0e43b00-f46a-11e8-9810-1395e9fc6afb.png)


As you can see, the first and the second 'Blockquotes' are the same. That's because `p:nth-child` counts every paragraph not only these with the blocquotes inside and as it's presented in the example there are edge cases when two or more following Pullquotes will have the same styles.

>  If the idea is that we want to track instances/occurrences, we can use a local variable (a side-effect)

I like it :+1: 

> navigating to a second page would start on the 2nd style. 

It's not a problem. What matters is that every following Pullquote will have different style, doesn't matter how many generic `<p>` tags will be between it and the previous one. I can refactor the code this way 

> I want to give you some options, [...] just want to make sure we at least vet some other options!

I like it too :100:

Reply:
@DSchau I refactored code, it's look much cleaner now. Thank you  Dustin! :1st_place_medal: